File: Warehouse.Web/Models/Tenant/Project/Job/JobPriority.cs
using System;
using System.Collections;
using System.Collections.Generic;
using Warehouse.Models.Joins;

namespace Warehouse.Models
{
    public class JobPriority
    {
        public Guid Id { get; set; }
        public int Order { get; set; }
        public string Name { get; set; }
        public string Colour { get; set; }
        public IList<Job> Jobs { get; set; }
    }
    
    public class NewPriority
    {
        public string Name { get; set; }
        public string Colour { get; set; }
    }
    
    public class JobPriorityViewModel
    {
        public Guid Id { get; set; }
        public int Order { get; set; }
        public string Name { get; set; }
        public string Colour { get; set; }
    }
}

File: Warehouse.Web/ClientApp/src/app/Models/JobPriority.ts
import {User} from "./User";

export interface JobPriority {
    id: string;
    name: string;
    colour: string;
    order: number;
    user: User | null;
}

export interface NewPriority
{
  name: string;
  order: number;
  colour: string;
}
